src/hdmf/container.py

```diff
@@ -728,18 +730,22 @@ def _generate_field_html(self, key, value, level, access_code):
             return f'<div style="margin-left: {level * 20}px;" class="container-fields"><span class="field-key"' \
                    f' title="{access_code}">{key}: </span><span class="field-value">{value}</span></div>'
 
-        if hasattr(value, "generate_html_repr"):
+        if isinstance(value, (np.ndarray, h5py.Dataset)):
+            html_content = self._generate_array_html(value, level + 1)
+        elif hasattr(value, "store") and hasattr(value, "shape"):  # Duck typing for zarr array
+            html_content = self._generate_array_html(value, level + 1)
+        elif hasattr(value, "generate_html_repr"):
             html_content = value.generate_html_repr(level + 1, access_code)
-
         elif hasattr(value, '__repr_html__'):
             html_content = value.__repr_html__()
-
         elif hasattr(value, "fields"):
```

Comment:
I think we maybe should add a comment here. I think this branching was intended for Containers that have fields but it also captures h5py.Datasets because they have a method called fields. I am solving this by just having the branching to catch h5py.Datasets before this but it seems brittle to rely on order of execution.

Reply:
Done.